src/OrchardCore/Orchard.Environment.Extensions/ExtensionManager.cs

```diff
@@ -90,6 +90,11 @@ public class ExtensionManager : IExtensionManager
 
         public IExtensionInfo GetExtension(string extensionId)
         {
+            if (string.IsNullOrEmpty(extensionId))
```

Comment:
If you throw an exception, it probably will not help. As a matter of fact the exception is thrown down the stack when a null area object is referenced, and this exception terminates pipeline processing. As a result the controller is never reached.